backend/templates/import.html

```diff
@@ -0,0 +1,222 @@
+      e.stopPropagation();
+      e.preventDefault();
+      $(this).addClass('hover');
+    });
+    holder.on('dragleave', function (e) {
+      e.preventDefault();
+      $(this).removeClass('hover');
+    });
+  });
+
+  // show upload wait
+  function start_wait() {
+    $('#start-upload-dialog').modal('hide');
+    $('.ui.page.dimmer').addClass('active');
+  }
+
+  // stop upload wait
+  function stop_wait(msg, success) {
+    $('.ui.page.dimmer').removeClass('active');
+    $('#upload-dialog-msg').html(msg);
+    $('#upload-dialog').modal({
+      closable: false, // protect closing of uploading dialog
+      onDeny() {
+        return false;
+      },
+      onHide() {
+        return false;
+      }
+    }).modal('show');
+
+    if (success) {
+      $('#upload-done-button').addClass('positive')
+    } else {
+      $('#upload-done-button').addClass('red')
+    }
+  }
+
+  // send data with tasks to server (add more data)
+  function send_data(event, upload_url = '') {
+    if (upload_url === '' && typeof event.target.files === 'undefined') {
+      return;
+    }
+
+    // show upload wait dialog
+    start_wait();
+    let fd = null;
+    let request = {};
+
+    // send url to file
+    if (upload_url !== '') {
+      // send url as task data
+      if (IsJsonString(upload_url)) {
+        request = {
+          url: "api/import",
+          data: upload_url,
+          contentType: 'application/json',
+          method: 'post'
+        }
+      }
+      // send url as regular url
+      else {
+        request = {
+          url: "api/import",
+          data: {url: upload_url},
+          method: 'post'
+        }
+      }
+    }
+
+    // files from disk
+    else {
+      fd = new FormData;
+      for (let i = 0; i < event.target.files.length; i++) {
+
+        const f = event.target.files[i];
+        let max_size_mb = {{ project.max_tasks_file_size }};
+        if (f.size / 1024.0 / 1024.0 > max_size_mb) {
+          stop_wait('Sorry, but file size is too big (more ' + max_size_mb + ' mb).\n<br/>' +
+            'Try to split your file by chunks or zip it', false);
+        }
+
+        // make form and attach file to it
+        fd.append(f.name, f);
+      }
+
+      request = {
+        url: "api/import",
+        data: fd,
+        method: 'POST',
+        processData: false,
+        contentType: false
+      }
+    }
+
+    $.ajax(request)
+      .done(answer => {
+        let msg = 'Tasks created: ' + answer['task_count'] +
+          '<br/>Completions created: ' + answer['completion_count'] +
+          '<br/>Predictions created: ' + answer['prediction_count'] +
+          '<br/>Duration: ' + answer['duration'].toFixed(2) + ' sec';
+        stop_wait(msg, true);
+
+      })
+      .fail(answer => {
+        let msg = "Error: can't upload/process file on server side. Reasons:<br><br>";
+
+        if (answer.responseJSON != null) {
+          let rows = answer.responseJSON;
+          for (let i in rows) {
```

Comment:
May be shorten this cycle to 3-5 items?

Reply:
It will cause differences in django python code. I want to avoid this.